### DISCUSSION TITLE
Migrations never run blocking deployment

### DISCUSSION THREAD

**Discussion by cablespaghetti** [apache/airflow#32012] [General]
### Official Helm Chart version

1.9.0 (latest released)

### Apache Airflow version

2.6.2

### Kubernetes Version

1.25.9

### Helm Chart configuration

All relevant settings for this issue are default. My database was on v2.5 and I was trying to upgrade to v2.6. This doesn't seem to impact upgrades or fresh deployments if the database is already on the right version.

### What happened

The migrations job has is set to run `post-install,post-upgrade` but because the rest of the Pods have init containers to `wait-for-migrations` the post-install/upgrade hooks never run. This results in the chart never finishing deploying.

https://github.com/apache/airflow/blob/0560881f0eaef9c583b11e937bf1f79d13e5ac7c/chart/templates/jobs/migrate-database-job.yaml#LL45C1-L45C77

I've worked around this by tweaking the annotations:
```
    migrateDatabaseJob:
      useHelmHooks: false
      annotations:
        helm.sh/hook: "pre-install,pre-upgrade"
        helm.sh/hook-weight: "1"
```

### What you think should happen instead

The migrations should be a pre-upgrade/pre-install hook to ensure they have run before the rest of the Pods come up.

### How to reproduce

Try upgrading from 2.5 to 2.6 and see the pods wait-for-migrations forever and migrations never running. This might impact fresh installs, I am not certain yet.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by hussein-awala** [comment]
IMHO, this job should be run after the chart is installed, where it initialize/update the database which could be a part of the chart. The other pods have a waiting pod to wait it, so there is no problem with that, but if we run it with `pre-install`, and with `postgresql.enabled=true`, there will be a problem in some cases.

I convert the issue to discussion, and we can reopen it if needed.

cc: @jedcunningham

**Comment by jedcunningham** [comment]
Do you happen to be using `--wait` with helm? The short answer is if you want to use `--wait`, you need to disable the hook behavior like you discovered. That is expected and [documented here](https://airflow.apache.org/docs/helm-chart/stable/index.html#installing-the-chart-with-argo-cd-flux-rancher-or-terraform).
